internal/service/resiliencehub/app.go

```diff
@@ -0,0 +1,1502 @@
+		return
+	}
+
+	conn := r.Meta().ResilienceHubClient(ctx)
+
+	name := plan.Name.ValueString()
+	input := &resiliencehub.CreateAppInput{
+		Tags: getTagsIn(ctx),
+	}
+
+	// Use AutoFlex to expand plan to input
+	resp.Diagnostics.Append(flex.Expand(ctx, plan, input)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	output, err := conn.CreateApp(ctx, input)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionCreating, ResNameApp, name, err),
+			err.Error(),
+		)
+		return
+	}
+
+	plan.ID = types.StringValue(aws.ToString(output.App.AppArn))
+
+	// Use AutoFlex to flatten the created app back to plan
+	resp.Diagnostics.Append(flex.Flatten(ctx, output.App, &plan, flex.WithNoIgnoredFieldNames())...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Put app template
+	if !plan.AppTemplate.IsNull() && !plan.AppTemplate.IsUnknown() {
+		templateBody, diags := r.expandAppTemplate(ctx, plan.AppTemplate)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		templateInput := &resiliencehub.PutDraftAppVersionTemplateInput{
+			AppArn:          output.App.AppArn,
+			AppTemplateBody: aws.String(templateBody),
+		}
+
+		_, err = conn.PutDraftAppVersionTemplate(ctx, templateInput)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionCreating, ResNameApp, name, err),
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	// Import resources with full mapping support
+	if !plan.ResourceMapping.IsNull() && !plan.ResourceMapping.IsUnknown() {
+		sourceArns, terraformSources, diags := r.expandResourceMappingsToSources(ctx, plan.ResourceMapping)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		// Only call ImportResourcesToDraftAppVersion if we have actual sources to import
+		if len(sourceArns) > 0 || len(terraformSources) > 0 {
+			importInput := &resiliencehub.ImportResourcesToDraftAppVersionInput{
+				AppArn:           output.App.AppArn,
+				SourceArns:       sourceArns,
+				TerraformSources: terraformSources,
+			}
+
+			_, err := conn.ImportResourcesToDraftAppVersion(ctx, importInput)
+			if err != nil {
+				resp.Diagnostics.AddError(
+					create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionCreating, ResNameApp, name, err),
+					err.Error(),
+				)
+				return
+			}
+		}
+	}
+
+	createTimeout := r.CreateTimeout(ctx, plan.Timeouts)
+	app, err := waitAppCreated(ctx, conn, plan.ID.ValueString(), createTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.ResilienceHub, create.ErrActionWaitingForCreation, ResNameApp, name, err),
+			err.Error(),
+		)
+		return
+	}
+
+	// Handle drift status - ensure it's never unknown
+	if app.DriftStatus != "" {
+		plan.DriftStatus = types.StringValue(string(app.DriftStatus))
+	} else {
+		plan.DriftStatus = types.StringValue("NotChecked")
+	}
```

Comment:
The value of `drift_status` should be set to whatever the AWS API returns. It will never be unknown

```suggestion
	plan.DriftStatus = types.StringValue(string(app.DriftStatus))
```